test/e2e/e2e_test.go

```diff
@@ -38,6 +38,86 @@ import (
 
 // +kubebuilder:docs-gen:collapse=Imports
 
+var _ = ginkgo.Describe("Kueue visibility server", func() {
+	const defaultFlavor = "default-flavor"
+	var (
+		defaultRF    *kueue.ResourceFlavor
+		localQueue   *kueue.LocalQueue
+		clusterQueue *kueue.ClusterQueue
+		ns           *corev1.Namespace
+		sampleJob1   *batchv1.Job
+		sampleJob2   *batchv1.Job
+	)
+
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "e2e-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+
+		defaultRF = testing.MakeResourceFlavor(defaultFlavor).Obj()
+		gomega.Expect(k8sClient.Create(ctx, defaultRF)).Should(gomega.Succeed())
+
+		clusterQueue = testing.MakeClusterQueue("cluster-queue").
+			ResourceGroup(
+				*testing.MakeFlavorQuotas(defaultFlavor).
+					Resource(corev1.ResourceCPU, "1").
+					Obj(),
+			).Obj()
+		gomega.Expect(k8sClient.Create(ctx, clusterQueue)).Should(gomega.Succeed())
+
+		localQueue = testing.MakeLocalQueue("main", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+		gomega.Expect(k8sClient.Create(ctx, localQueue)).Should(gomega.Succeed())
+	})
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		gomega.Expect(util.DeleteLocalQueue(ctx, k8sClient, localQueue)).Should(gomega.Succeed())
+		util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueue, true)
+		util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, defaultRF, true)
+	})
+
+	ginkgo.When("Creating two Jobs with max capacity", func() {
+		ginkgo.BeforeEach(func() {
+			sampleJob1 = testingjob.MakeJob("test-job-1", ns.Name).
+				Queue(localQueue.Name).
+				Image("gcr.io/k8s-staging-perf-tests/sleep:v0.0.3", []string{"3s"}).
+				Request(corev1.ResourceCPU, "1").
+				Obj()
+			sampleJob2 = testingjob.MakeJob("test-job-2", ns.Name).
+				Queue(localQueue.Name).
+				Image("gcr.io/k8s-staging-perf-tests/sleep:v0.0.3", []string{"100ms"}).
+				Request(corev1.ResourceCPU, "1").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, sampleJob1)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, sampleJob2)).Should(gomega.Succeed())
+		})
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteAllJobsInNamespace(ctx, k8sClient, ns)).Should(gomega.Succeed())
+		})
+
+		ginkgo.It("Should fetch information about pending workloads", func() {
```

Comment:
I'm a little bit worried about the 3s runtime of a job, if the test could flake on some heavily loaded environments. 

Maybe we could have a little more confidence in stability with the following scenario:
- verify the pending workloads are empty
- schedule `sampleJob1` which runs forever, it gets admitted and blocks quota
- schedule `sampleJob2` which is queued as pending
- verify we have a single job pending
- delete `sampleJob1` by delete request, so that if frees the resources, allowing for `sampleJob2` to get admitted
- verify the list of pending workloads is empty once `sampleJob2` is admitted